wlauto/workloads/uxperf_applaunch/__init__.py

```diff
@@ -0,0 +1,168 @@
+                  mandatory=True),
+        Parameter('workload_params', kind=dict, default={},
+                  description="""
+                  parameters of the uxperf workload whose application launch
+                  time is measured
+                  """),
+        Parameter('applaunch_type', kind=str, default='launch_from_background',
+                  allowed_values=['launch_from_background', 'launch_from_long-idle'],
+                  description="""
+                  Choose launch_from_long-idle for measuring launch time
+                  from long-idle. These two types are described in the class
+                  description.
+                  """),
+        Parameter('applaunch_iterations', kind=int, default=1,
+                  description="""
+                  Number of iterations of the application launch
+                  """),
+    ]
+    
+    #Path where the workload jar file will be pushed on device
+    workload_jar_path = "/data/local/tmp" 
+
+    def __init__(self, device, **kwargs):
+        super(UxperfApplaunch, self).__init__(device, **kwargs)
+        loader = ExtensionLoader(packages=settings.extension_packages, paths=settings.extension_paths)
+        self.workload_params['markers_enabled'] = self.markers_enabled
+        self.workload = loader.get_workload(self.workload_name, device,
+                                            **self.workload_params)
+
+    def init_resources(self, context):
+        super(UxperfApplaunch, self).init_resources(context)
+        self.init_workload_resources(context)
+
+    def init_workload_resources(self, context):
+        self.workload.uiauto_file = context.resolver.get(wlauto.common.android.resources.JarFile(self.workload))
+        print self.workload.uiauto_file
+        if not self.workload.uiauto_file:
+            raise ResourceError('No UI automation JAR file found for workload {}.'.format(self.workload.name))
+        self.workload.device_uiauto_file = self.device.path.join(self.workload_jar_path,
+                                                        os.path.basename(self.workload.uiauto_file))
+        print self.workload.device_uiauto_file
+        if not self.workload.uiauto_package:
+            self.workload.uiauto_package = os.path.splitext(os.path.basename(self.workload.uiauto_file))[0]
+
+    def validate(self):
+        super(UxperfApplaunch, self).validate()
+        self.workload.validate()
+        self.pass_parameters()
+
+    def pass_parameters(self):
+        self.uiauto_params['workload'] = self.workload.name
+        self.uiauto_params['workloadJarPath'] = self.workload_jar_path
+        self.uiauto_params['package'] = self.workload.package
+        self.uiauto_params.update(self.workload.uiauto_params)
+        if self.workload.activity:
+            self.uiauto_params['launch_activity'] = self.workload.activity
+        else:
+            self.uiauto_params['launch_activity'] = "None"
```

Comment:
If this is being checked and set here is the previous setting of `activity` needed?